zingocli/tests/integration_tests.rs

```diff
@@ -1086,6 +1087,73 @@ async fn send_to_ua_saves_full_ua_in_wallet() {
     drop(child_process_handler);
 }
 
+#[tokio::test]
+async fn self_send_to_t_displays_as_one_transaction() {
+    let (regtest_manager, child_process_handler, faucet, recipient) =
+        scenarios::faucet_recipient().await;
+    let recipient_unified_address = get_base_address!(recipient, "unified");
+    let sent_value = 50_000;
+    faucet
+        .do_send(vec![(recipient_unified_address.as_str(), sent_value, None)])
+        .await
+        .unwrap();
+    utils::increase_height_and_sync_client(&regtest_manager, &recipient, 1).await;
+    let recipient_taddr = get_base_address!(recipient, "transparent");
+    let recipient_zaddr = get_base_address!(recipient, "sapling");
+    let sent_to_taddr_value = 5_000;
+    let sent_to_zaddr_value = 11_000;
+    let sent_to_self_orchard_value = 1_000;
+    recipient
+        .do_send(vec![(recipient_taddr.as_str(), sent_to_taddr_value, None)])
+        .await
+        .unwrap();
+    utils::increase_height_and_sync_client(&regtest_manager, &recipient, 1).await;
+    recipient
+        .do_send(vec![
+            (recipient_taddr.as_str(), sent_to_taddr_value, None),
+            (
+                recipient_zaddr.as_str(),
+                sent_to_zaddr_value,
+                Some("foo".to_string()),
+            ),
+            (
+                recipient_unified_address.as_str(),
+                sent_to_self_orchard_value,
+                Some("bar".to_string()),
+            ),
+        ])
+        .await
+        .unwrap();
+    faucet.do_sync(false).await.unwrap();
+    faucet
+        .do_send(vec![
+            (recipient_taddr.as_str(), sent_to_taddr_value, None),
+            (
+                recipient_zaddr.as_str(),
+                sent_to_zaddr_value,
+                Some("foo2".to_string()),
+            ),
+            (
+                recipient_unified_address.as_str(),
+                sent_to_self_orchard_value,
+                Some("bar2".to_string()),
+            ),
+        ])
+        .await
+        .unwrap();
+    utils::increase_height_and_sync_client(&regtest_manager, &recipient, 1).await;
+    println!(
+        "{}",
+        json::stringify_pretty(recipient.do_list_transactions(false).await, 4)
+    );
+    let transactions = recipient.do_list_transactions(false).await;
+    let mut txids = transactions
+        .members()
+        .map(|transaction| transaction["txid"].as_str());
+    assert!(txids.all_unique());
```

Comment:
Is this an `iter_tools` trait?   Is this the only reason we use `iter_tools`?

In the case of a unique (or very rare) use I find it easier to read explicitly qualified names.   Since explicit qualification isn't more verbose than `use` statements, when the use is rare, I vote for that pattern.